contracts/oracles/TwapOracle.sol

```diff
@@ -63,6 +56,12 @@ contract TwapOracle is OwnableUpgradeable, TwapInterface {
         uint256 newAcc
     );
 
+    ///@notice Keeps track of observation of token mapped by address , update on evry updateTwap invocation.
+    mapping(address => Observation[]) public observations;
```

Comment:
Can we make this `private` ?

Reply:
This should not be necessary IMO as this is not any kind of sensitive data. This is kind of important for our test cases to analyze if the observations are getting updated properly. Also, on deployment, it will be easier to debug the observations in case there is some issue. If not public we will have to make a view function.

Reply:
private is better in terms of gas optimisation. In our tests we can use a scenario contract that inherits the TWAP contract so that we can read the private variables.